pkgs/development/compilers/gnat-bootstrap/default.nix

```diff
@@ -38,15 +47,24 @@ let
 in with versionMap.${majorVersion};
 
 stdenv.mkDerivation rec {
-  pname = "gnatboot";
-  inherit gccVersion alireRevision;
+  pname = "gnat-bootstrap";
+  inherit gccVersion;
 
-  version = "${gccVersion}-${alireRevision}";
-
-  src = fetchzip {
-    url = "https://github.com/alire-project/GNAT-FSF-builds/releases/download/gnat-${version}/gnat-${stdenv.hostPlatform.system}-${version}.tar.gz";
-    inherit hash;
-  };
+  version =
+    if majorVersion == "4" then
+      gccVersion
+    else
+      "${gccVersion}-${alireRevision}";
+  src =
+    if majorVersion == "4" then
+      fetchurl {
+        inherit url hash;
+      }
+    else
+      fetchzip {
+        url = "https://github.com/alire-project/GNAT-FSF-builds/releases/download/gnat-${version}/gnat-${stdenv.hostPlatform.system}-${version}.tar.gz";
+        inherit hash;
+      };
```

Comment:
Maybe we can move this into versionMap, to reduce the branching and make the code a bit cleaner.

Reply:
I tried this there too originally, the `version` is easily moved up and I'll add that in. On the other hand `src` is quite a pain. I'm not sure how I'd avoid repeating the alire-project releases url for each version and architecture, it doesn't seem to end up being cleaner at all for `src`.

Reply:
Actually that doesn't work either, not sure why. When I tried building with cb798bcae40 nix-build errored with:
```
nix-build -A gnat-bootstrap4
error: undefined variable 'gccVersion'

       at /home/manny/cloned/github.com/NixOS/nixpkgs/pkgs/development/compilers/gnat-bootstrap/default.nix:23:20:

           22|       alireRevision = "4";
           23|       version = "${gccVersion}-${alireRevision}";
             |                    ^
           24|     } // {
```

So going to go back.